client/controller/web-util.js

```diff
@@ -22,6 +24,20 @@ export function redirectLoggedIn( context, next ) {
 
 export function render( context ) {
 	ReactDom.render( context.layout, document.getElementById( 'wpcom' ) );
+
+	// Activate the next section after it's been successfully loaded.
+	// (`context.nextSectionToActivate` is set in client/sections-middleware.js)
+	// Section activation is triggered when the new section is rendered for the
+	// first time; this avoids glitches caused by the section's styles loading
+	// before the actual section's markup gets rendered..
+	if ( context.nextSectionToActivate ) {
+		context.store.dispatch( setSection( context.nextSectionToActivate ) );
+		context.store.dispatch( activateNextLayoutFocus() );
+
+		// Removing `nextSectionToActivate` makes sure that `setSection` won't
+		// be dispatched until a new section gets activated.
+		delete context.nextSectionToActivate;
```

Comment:
Usually, I wouldn't recommend mutations, but it's not a big deal for `context`, because we're mutating it literally all around the place.

Reply:
Agreed, I had a look at the rest of the code for the section controllers and got "inspired" to go for this approach 😅

Reply:
The lifetime of the `context` objects lasts only until the current _page.js_ navigation finishes. And that navigation is guaranteed to finish right after the `render` function returns, as it doesn't call `next()`. Deleting the property is therefore not necessary.

Reply:
That's a good point, @jsnajdr — will remove those lines